internal/provider/common_cert.go

```diff
@@ -88,6 +88,7 @@ func generateSubjectKeyID(pubKey crypto.PublicKey) ([]byte, error) {
 			err = fmt.Errorf("received 'nil' pointer instead of public key")
 		}
 	case *ecdsa.PublicKey:
+		//nolint:staticcheck // Although Marshal is deprecated we cannot just switch to using p = pub.ECDH(), p.Bytes()
 		pubKeyBytes = elliptic.Marshal(pub.Curve, pub.X, pub.Y)
```

Comment:
I have little to no domain knowledge of the Go function being deprecated here 😞 . But in general I feel like if this needs more research to determine how best to replace this deprecated function, we should do it outside of this PR. (it doesn't feel like an urgent thing, maybe a gardening thing 🧑‍🌾 )

Can you dump any context you may have about the deprecation into a new GH issue and link it in this `nolint` comment?

Reply:
Have added a note linking to https://github.com/hashicorp/terraform-provider-tls/issues/480